app/models/ability.rb

```diff
@@ -2,31 +2,21 @@ class Ability
   include CanCan::Ability
 
   def initialize(user)
-    # Define abilities for the passed in user here. For example:
-    #
-    #   user ||= User.new # guest user (not logged in)
-    #   if user.admin?
-    #     can :manage, :all
-    #   else
-    #     can :read, :all
-    #   end
-    #
-    # The first argument to `can` is the action you are giving the user
-    # permission to do.
-    # If you pass :manage it will apply to every action. Other common actions
-    # here are :read, :create, :update and :destroy.
-    #
-    # The second argument is the resource the user can perform the action on.
-    # If you pass :all it will apply to every resource. Otherwise pass a Ruby
-    # class of the resource.
-    #
-    # The third argument is an optional hash of conditions to further filter the
-    # objects.
-    # For example, here the user can only update published articles.
-    #
-    #   can :update, Article, :published => true
-    #
-    # See the wiki for details:
-    # https://github.com/ryanb/cancan/wiki/Defining-Abilities
+    # user ||= User.new
+    if user.is_a?(PartneringOrganization)
+      user ||= PartneringOrganization.new
+      can :read, PartneringOrganization
+      can :manage, PartneringOrganization
+    elsif user.is_a?(Veteran)
+      user ||= Veteran.new
+      can :read, Veteran
+      can :manage, Veteran
```

Comment:
this line is kinda sketchy cause I think it also implies that a veteran can manage another veteran's profile that is not him/herself